### PR TITLE
Add device pixel ratio to MediaQuery

### PR DESCRIPTION
**Pull request by krisgiesing**

### CODE REVIEW

File: packages/flutter/lib/src/widgets/binding.dart [hunk 1]
```diff
@@ -62,7 +62,7 @@ class WidgetFlutterBinding extends BindingBase with Scheduler, Gesturer, Rendere
   void handleMetricsChanged() {
     super.handleMetricsChanged();
     for (BindingObserver observer in _observers)
-      observer.didChangeSize(ui.window.size);
+      observer.didChangeMetrics(ui.window.size);
```
Comment (abarth) [inline packages/flutter/lib/src/widgets/binding.dart]: Probably should skip argument here.

File: packages/flutter/lib/src/widgets/media_query.dart [hunk 2]
```diff
@@ -16,11 +16,16 @@ enum Orientation {
...
-  const MediaQueryData({ this.size, this.padding });
+  const MediaQueryData({ this.size, this.devicePixelRatio, this.padding });
 
   /// The size of the media (e.g, the size of the screen).
   final Size size;
 
+  /// The number of device pixels for each logical pixel. This number might not
+  /// be a power of two. Indeed, it might not even be an integer. For example,
+  /// the Nexus 6 has a device pixel ratio of 3.5.
+  final double devicePixelRatio;
```
Comment (abarth) [inline packages/flutter/lib/src/widgets/media_query.dart]: You need to update operator== and hashCode as well.

**Comment by abarth** [comment]
LGTM

File: packages/flutter/lib/src/widgets/media_query.dart (hunk 2 above)
Comment (krisgiesing) [inline packages/flutter/lib/src/widgets/media_query.dart]: Done

**Pull request merged**